sgkit/stats/aggregation.py

```diff
@@ -285,6 +285,143 @@ def call_rate(ds: Dataset, dim: Dimension, call_genotype_mask: Hashable) -> Data
     )
 
 
+@numba_jit
+def _greatest_common_denominator(x: int, y: int) -> int:
+    while y != 0:
+        t = x % y
+        x = y
+        y = t
+    return x
+
+
+# TODO: replace with math.comb when supported by numba
+@numba_jit
+def _comb(n: int, k: int) -> int:
+    if k > n:
+        return 0  # pragma: no cover
+    r = 1
+    for d in range(1, k + 1):
+        gcd = _greatest_common_denominator(r, d)
+        r //= gcd
+        r *= n
+        r //= d // gcd
+        n -= 1
+    return r
+
+
+@numba_jit
+def _comb_with_replacement(n: int, k: int) -> int:
+    n = n + k - 1
+    return _comb(n, k)
+
+
+@numba_guvectorize(  # type: ignore
+    [
+        "void(int8[:,:], uint64[:], uint64[:])",
+        "void(int16[:,:], uint64[:], uint64[:])",
+        "void(int32[:,:], uint64[:], uint64[:])",
+        "void(int64[:,:], uint64[:], uint64[:])",
+    ],
+    "(n, k),(g)->(g)",
+)
+def _count_genotypes(
+    genotypes: ArrayLike, _: ArrayLike, out: ArrayLike
+) -> ArrayLike:  # pragma: no cover
+    # Warning: genotype alleles must be sorted in ascending order!
+    n, k = genotypes.shape
+    out[:] = 0
+    for i in range(n):
+        index = 0
+        compleat = True
+        for j in range(k):
+            a = genotypes[i, j]
+            if a >= 0:
+                index += _comb_with_replacement(a, j + 1)
+            else:
+                compleat = False
+        if compleat:
+            out[index] += 1
+
+
+def count_variant_genotypes(
+    ds: Dataset,
+    *,
+    call_genotype: Hashable = variables.call_genotype,
+    merge: bool = True,
+) -> Dataset:
+    """Count the number of calls for each possible genotype at each variant.
```

Comment:
Can you spell out what you mean by "genotype" here? It means different things to different people, and helpful to be precise about it somewhere.

Reply:
Hope it's clearer now